rtt_roscomm/include/rtt_roscomm/passthrough_callback_queue.hpp

```diff
@@ -8,224 +7,39 @@
-      void(T::*fp)(const boost::shared_ptr<M const>&) const, 
-      const boost::shared_ptr<T>& obj,
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template init<M>(topic, queue_size, boost::bind(fp, obj.get(), _1));
-    ops.tracked_object = obj;
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size, void(*fp)(M),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template initByFullCallbackType<M>(topic, queue_size, fp);
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size,
-      void(*fp)(const boost::shared_ptr<M const>&),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template init<M>(topic, queue_size, fp);
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size,
-      const boost::function<void (const boost::shared_ptr<M const>&)>& callback,
-      const ros::VoidConstPtr& tracked_object = ros::VoidConstPtr(),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template init<M>(topic, queue_size, callback);
-    ops.tracked_object = tracked_object;
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M, class C>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size,
-      const boost::function<void (C)>& callback,
-      const ros::VoidConstPtr& tracked_object = ros::VoidConstPtr(),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template initByFullCallbackType<C>(topic, queue_size, callback);
-    ops.tracked_object = tracked_object;
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
+class PassthroughCallbackQueue: public ros::CallbackQueueInterface
+{
+  public:
+    /** 
+     * Implementation of ros::CallbackQueueInterface::addCallback()
+     * 
+     * This method is executing the callback received instead of adding
+     * it to a queue. In this way, the queue is bypassed and immediately
+     * executed.
```

Comment:
```suggestion
     * it to a queue. In this way, the queue is bypassed and the callback is immediately
     * executed.
```

But actually that sentence is just repeating the first.